src/weapon/w_029.c

```diff
@@ -2,12 +2,143 @@
 // Fire shield, Unknown#213
 #include "weapon_private.h"
 #include "shared.h"
-
+extern s32 D_CF000_8017AC78;
+extern s32 D_CF000_8017ACC0;
 extern u16 D_CF000_8017ACF8[];
 extern u16 D_CF000_8017AD04[];
 extern AnimationFrame D_CF000_8017AD24[];
 
-INCLUDE_ASM("weapon/nonmatchings/w_029", EntityWeaponAttack);
+void EntityWeaponAttack(Entity* self) {
+    s32 newUnkAC;
+    s32 handButton;
+    s32 unkAC_offset;
+
+    newUnkAC = 0;
+    unkAC_offset = 0;
+    if (g_HandId != 0) {
+        handButton = PAD_CIRCLE;
+    } else {
+        handButton = PAD_SQUARE;
+    }
+    if (!(handButton & g_Player.padPressed) && (self->step < 3)) {
+        self->animFrameDuration = 0;
+        self->animFrameIdx = 0;
+        self->step = 3;
+    }
+    if (self->step != 4) {
+        self->posX.val = PLAYER.posX.val;
+        self->posY.val = PLAYER.posY.val;
+        self->facingLeft = PLAYER.facingLeft;
+    }
+    if ((g_Player.unk0C & 0x10000) && (self->step != 4)) {
+        self->zPriority = PLAYER.zPriority + 2;
+        self->step = 4;
+        if (g_Player.pl_vram_flag & 1) {
+            self->velocityX = PLAYER.velocityX;
+        } else {
+            self->velocityX = PLAYER.velocityX * 2;
+        }
+        self->velocityY = FIX(-3.5);
+        self->ext.weapon.lifetime = 128;
+        self->flags = FLAG_UNK_08000000;
+        self->animCurFrame = 0x3E;
+    }
+    if ((PLAYER.step == 2) && (PLAYER.step_s != 2)) {
+        unkAC_offset++;
+    }
+    switch (self->step) {
+    case 0:
+        SetSpriteBank1(g_Animset);
+        if (self->params & 0x8000) {
+            self->animSet = -0x7FEE;
```

Comment:
```
            self->animSet = 0x8012;
            self->ext.weapon.unk80 = 0x128;
            var_v0 = 0x66;
        } else {
            self->animSet = 0x8010;
```
We also have defines for these https://github.com/Xeeynamo/sotn-decomp/blob/bebb4789a139d322565206bd13ec930a8fcadadd/include/game.h#L191

Reply:
Oh fantastic, thanks for informing me! I think I did it properly now but let me know if anything is still amiss.